src/component.js

```diff
@@ -25,12 +25,19 @@ export function Component(props, context) {
  */
 Component.prototype.setState = function(update, callback) {
 	// only clone state when copying to nextState the first time.
-	let s =
-		(this._nextState !== this.state && this._nextState) ||
-		(this._nextState = assign({}, this.state));
+	let s;
+	if (this._nextState !== this.state) {
+		s = this._nextState;
+	}
+	else {
+		s = this._nextState = assign({}, this.state);
+	}
```

Comment:
Original:
```js
var u = (this.__s !== this.state && this.__s) || (this.__s = s({}, this.state));
```

Updated:
```js
var u;
u = this.__s !== this.state ? this.__s : (this.__s = s({}, this.state));
```

My guess is that since many of our minified functions begin with `var` statements the updated one compresses in gzip better. And perhaps the conversion to the ternary compresses better saving us an additional byte. Brotli stays the same size with this change